spec/requests/api/v1/moves_controller_filter_spec.rb

```diff
@@ -142,6 +142,36 @@
       end
     end
 
+    describe 'by has_relationship_to_allocation' do
+      let(:filter_params) { { filter: { has_relationship_to_allocation: filter_value } } }
+      let(:move_with_allocation) { create(:move, :with_allocation) }
+      let(:move_without_allocation) { create(:move) }
+
+      context 'when set to true' do
+        let(:expected_moves) { Move.where(id: move_with_allocation.id) }
+        let(:unexpected_moves) { Move.where(id: move_without_allocation.id) }
+        let(:filter_value) { true }
+
+        it_behaves_like 'it returns the correct moves'
+      end
+
+      context 'when set to false' do
+        let(:expected_moves) { Move.where(id: move_without_allocation.id) }
+        let(:unexpected_moves) { Move.where(id: move_with_allocation.id) }
+        let(:filter_value) { false }
+
+        it_behaves_like 'it returns the correct moves'
+      end
+
+      context 'when set to incorrect type' do
+        let(:expected_moves) { Move.where(id: [move_without_allocation.id, move_with_allocation.id]) }
+        let(:unexpected_moves) { Move.none }
+        let(:filter_value) { 'some-value' }
+
+        it_behaves_like 'it returns the correct moves'
```

Comment:
Does this need a test case for the key being excluded completely? Or set to `null` maybe?